src/pip/_internal/vcs/git.py

```diff
@@ -83,21 +88,14 @@ def is_immutable_rev_checkout(self, url, dest):
         )
         return not is_tag_or_branch
 
-    def get_git_version(self):
-        # type: () -> _BaseVersion
-        VERSION_PFX = 'git version '
+    def get_git_version(self) -> Tuple[int, ...]:
         version = self.run_command(
             ['version'], show_stdout=False, stdout_only=True
         )
-        if version.startswith(VERSION_PFX):
-            version = version[len(VERSION_PFX):].split()[0]
-        else:
-            version = ''
-        # get first 3 positions of the git version because
-        # on windows it is x.y.z.windows.t, and this parses as
-        # LegacyVersion which always smaller than a Version.
-        version = '.'.join(version.split('.')[:3])
-        return parse_version(version)
+        match = GIT_VERSION_REGEX.match(version)
+        if not match:
```

Comment:
I'm a little late to the game but would it make sense to have a `logger.debug` (or `logger.info/warning`) call here mentioning that the version couldn't be parsed ?

Reply:
A warning sounds right to me.

Reply:
Filed #10361 for this.